src/target/riscv/riscv-013.c

```diff
@@ -2550,8 +2553,9 @@ static int write_memory(struct target *target, target_addr_t address,
 		uint32_t size, uint32_t count, const uint8_t *buffer)
 {
 	RISCV013_INFO(info);
-	if (info->progbufsize >= 2)
+	if (info->progbufsize >= 2 && !riscv_prefer_sba)
```

Comment:
What if I prefer SBA and have no SBA support? I guess this fails then. It might be nice to make it work even if you preferred SBA but don't support SBA, but again this isn't expected to be used much.

Reply:
Then a few lines down (line 2570) the memory write will happen using the program buffer.

Reply:
ah gotcha, looks good.